lib/RAST_SDK/AnnotationUtils.pm

```diff
@@ -415,19 +415,48 @@ sub _check_NC_features {
     }
 }
 
-
 ## Mapping the contigIDs back to their original (long) names
 sub _remap_contigIDs {
-    my ($self, $contigID_hash, $genome) = @_;
+    my ($self, $contigID_hash, $gn) = @_;
+
+    return $gn unless $contigID_hash;
 
-    my $contigs = $genome->{contigs};
-    foreach my $ctg_obj (@{$contigs}) {
-        my $cid = $ctg_obj->{id};
-        $ctg_obj->{id} = $contigID_hash->{$cid};
-        $ctg_obj->{name} = $contigID_hash->{$cid};
+    if( $gn->{contig_ids} && @{$gn->{contig_ids}} > 0 ) {
+      for my $ctg_id (@{$gn->{contig_ids}}) {
+        $ctg_id = $contigID_hash->{ $ctg_id } if $contigID_hash->{ $ctg_id };
+      }
     }
-    return $genome;
+    if( $gn->{contigs} && @{$gn->{contigs}} > 0) {
+      for my $ctg (@{$gn->{contigs}}) {
+        $ctg->{id} = $contigID_hash->{$ctg->{id}} if $contigID_hash->{$ctg->{id}};
+        $ctg->{name} = $contigID_hash->{$ctg->{name}} if $contigID_hash->{$ctg->{name}};
+      }
+    }
+    $gn->{features} = $self->_map_location_contigIDs($contigID_hash, $gn->{features});
+    $gn->{non_coding_features} = $self->_map_location_contigIDs(
+                                        $contigID_hash, $gn->{non_coding_features});
+    $gn->{cdss} = $self->_map_location_contigIDs($contigID_hash, $gn->{cdss});
+    $gn->{mrnas} = $self->_map_location_contigIDs($contigID_hash, $gn->{mrnas});
```

Comment:
This can be factored out again:

```perl
for my $feature_type ( qw( features non_coding_features ... ) ) {
  $gn->{ $feature_type } = $self->_map_location_contigIDs( ... );
}
```

Whenever you have lines of very similar code, you should always be thinking: can I simplify this further?

Reply:
done